lib/shelley/src/Cardano/Wallet/Shelley.hs

```diff
@@ -333,7 +335,8 @@ serveWallet
                 poolDatabaseDecorator
                 poolsDbTracer
                 (Pool.defaultFilePath <$> dir)
-                (timeInterpreter nl)
+                (neverFails "withPoolsMonitoring never forecasts into the future"
+                    $ timeInterpreter nl)
```

Comment:
What about stake pool retirements :thinking:  ?

Reply:
Hm, yeah, not obvious. This is used only for
1. [`selectPoolProduction`](https://github.com/input-output-hk/cardano-wallet/blob/4df2ecc5d5259ce333de6b2dfa39b6ac51b17652/lib/core/src/Cardano/Pool/DB/Sqlite.hs#L898-L908) which was jormungandr-specific and we should remove
2. [rolling back stake distributions](https://github.com/input-output-hk/cardano-wallet/blob/4df2ecc5d5259ce333de6b2dfa39b6ac51b17652/lib/core/src/Cardano/Pool/DB/Sqlite.hs#L487), which I guess also is unused and should be removed

Changing `"withPoolsMonitoring"` in the reason to `"pool db"` might help — if we don't want to clean this up now — but I think a separate PR would be better.